### PR TITLE
Simplify testing logits and fix test typo

### PR DESCRIPTION
**Pull request by ycxioooong**
This PR fixes:
1. Simplify testing logits by merging the case when tmpdir=None and specify tmpdir
2. Fix testing typo

### CODE REVIEW

File: mmcls/apis/test.py [hunk 1]
```diff
@@ -134,11 +134,12 @@ def collect_results_cpu(result_part, size, tmpdir=None):
         dist.broadcast(dir_tensor, 0)
         tmpdir = dir_tensor.cpu().numpy().tobytes().decode().rstrip()
     else:
-        if osp.exist(tmpdir):
-            raise OSError((f'The tmpdir {tmpdir} already exists.',
-                           ' Since tmpdir will be deleted after testing,',
-                           ' please make sure you specify an empty one.'))
-        mmcv.mkdir_or_exist(tmpdir)
+        if rank == 0:
+            if osp.exists(tmpdir):
```
Comment (mzr1996) [inline mmcls/apis/test.py:138]: A little suggestion, maybe we can combine two condition in one line.

**Review by mzr1996 — CHANGES_REQUESTED** [review]

File: mmcls/apis/test.py [hunk 2]
```diff
@@ -115,26 +114,10 @@ def multi_gpu_test(model, data_loader, tmpdir=None, gpu_collect=False):
...
-            tmpdir = tempfile.mkdtemp(dir='.dist_test')
-            tmpdir = torch.tensor(
-                bytearray(tmpdir.encode()), dtype=torch.uint8, device='cuda')
-            dir_tensor[:len(tmpdir)] = tmpdir
-        dist.broadcast(dir_tensor, 0)
-        tmpdir = dir_tensor.cpu().numpy().tobytes().decode().rstrip()
-    else:
-        if osp.exist(tmpdir):
+    if rank == 0:
+        if tmpdir != '.dist_test_mmcls' and osp.exists(tmpdir):
```
Comment (LXXXXR) [inline mmcls/apis/test.py:120]: May I ask what is `tmpdir != '.dist_test_mmcls'` for?
